AssetTracking/AssetTracking/Helpers/GraphService.cs

```diff
@@ -0,0 +1,189 @@
+                        return JsonConvert.SerializeObject(new { Message = $"The requested user '{email}' is invalid." }, Formatting.Indented);
+
+                    case "AuthenticationFailure":
+                        return JsonConvert.SerializeObject(new { e.Error.Message }, Formatting.Indented);
+
+                    case "TokenNotFound":
+                        await httpContext.ChallengeAsync();
+                        return JsonConvert.SerializeObject(new { e.Error.Message }, Formatting.Indented);
+
+                    default:
+                        return JsonConvert.SerializeObject(new { Message = "An unknown error has occurred." }, Formatting.Indented);
+                }
+            }
+        }
+
+        // Load user's profile picture in base64 string.
+        public static async Task<string> GetPictureBase64(GraphServiceClient graphClient, string email, HttpContext httpContext)
+        {
+            try
+            {
+                // Load user's profile picture.
+                var pictureStream = await GetPictureStream(graphClient, email, httpContext);
+
+                // Copy stream to MemoryStream object so that it can be converted to byte array.
+                var pictureMemoryStream = new MemoryStream();
+                await pictureStream.CopyToAsync(pictureMemoryStream);
+
+                // Convert stream to byte array.
+                var pictureByteArray = pictureMemoryStream.ToArray();
+
+                // Convert byte array to base64 string.
+                var pictureBase64 = Convert.ToBase64String(pictureByteArray);
+
+                return "data:image/jpeg;base64," + pictureBase64;
+            }
+            catch (Exception e)
+            {
+                switch (e.Message)
+                {
+                    case "ResourceNotFound":
+                        // If picture not found, return the default image.
+                        return "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4NCjwhRE9DVFlQRSBzdmcgIFBVQkxJQyAnLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4nICAnaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkJz4NCjxzdmcgd2lkdGg9IjQwMXB4IiBoZWlnaHQ9IjQwMXB4IiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDMxMi44MDkgMCA0MDEgNDAxIiB2ZXJzaW9uPSIxLjEiIHZpZXdCb3g9IjMxMi44MDkgMCA0MDEgNDAxIiB4bWw6c3BhY2U9InByZXNlcnZlIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPg0KPGcgdHJhbnNmb3JtPSJtYXRyaXgoMS4yMjMgMCAwIDEuMjIzIC00NjcuNSAtODQzLjQ0KSI+DQoJPHJlY3QgeD0iNjAxLjQ1IiB5PSI2NTMuMDciIHdpZHRoPSI0MDEiIGhlaWdodD0iNDAxIiBmaWxsPSIjRTRFNkU3Ii8+DQoJPHBhdGggZD0ibTgwMi4zOCA5MDguMDhjLTg0LjUxNSAwLTE1My41MiA0OC4xODUtMTU3LjM4IDEwOC42MmgzMTQuNzljLTMuODctNjAuNDQtNzIuOS0xMDguNjItMTU3LjQxLTEwOC42MnoiIGZpbGw9IiNBRUI0QjciLz4NCgk8cGF0aCBkPSJtODgxLjM3IDgxOC44NmMwIDQ2Ljc0Ni0zNS4xMDYgODQuNjQxLTc4LjQxIDg0LjY0MXMtNzguNDEtMzcuODk1LTc4LjQxLTg0LjY0MSAzNS4xMDYtODQuNjQxIDc4LjQxLTg0LjY0MWM0My4zMSAwIDc4LjQxIDM3LjkgNzguNDEgODQuNjR6IiBmaWxsPSIjQUVCNEI3Ii8+DQo8L2c+DQo8L3N2Zz4NCg==";
```

Comment:
Consider having the default photo stored in-memory in a folder and retrieved dynamically rather than having it string hard-coded in case one might want to change the default, and also to reduce the potential of a corrupted value.